### PR TITLE
Add Extra tags to karpenter config

### PR DESCRIPTION
**Pull request by DeeAjayi**

### CODE REVIEW

**Review by sachincool — APPROVED** [review]

File: charts/tfy-karpenter-config/templates/controlplane/karpenter-control-plane-ec2nodeclass.yaml [hunk 1]
```diff
@@ -23,6 +23,8 @@ spec:
         {{ $key }}: {{ $val | quote }} 
         {{- end }}
   tags:
+    cpu: "true"
```
Comment (sachincool) [inline charts/tfy-karpenter-config/templates/controlplane/karpenter-control-plane-ec2nodeclass.yaml:26]: we should rely on a tag, which can take both values. and then use filter on top, 
example: `compute_type: cpu|gpu`

Comment (shubhamrai1993) [reply charts/tfy-karpenter-config/templates/controlplane/karpenter-control-plane-ec2nodeclass.yaml:26]: +1

File: charts/tfy-karpenter-config/templates/default/karpenter-default-ec2nodeclass.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ spec:
         {{ $key }}: {{ $val | quote }} 
         {{- end }}
   tags:
+    cpu: "true"
```
Comment (shubhamrai1993) [inline charts/tfy-karpenter-config/templates/default/karpenter-default-ec2nodeclass.yaml:25]: can't we just use the extraTags field already present and only make the change in our env. Hardcoding the tags increases the chance of conflict with existing setup at customer's end

**Review by shubhamrai1993 — APPROVED** [review]

File: charts/tfy-karpenter-config/templates/default/karpenter-default-ec2nodeclass.yaml (hunk 2 above)
Comment (shubhamrai1993) [reply charts/tfy-karpenter-config/templates/default/karpenter-default-ec2nodeclass.yaml:25]: another option is to use truefoundry.com/compute_type as the tag key

**Review by shubhamrai1993 — APPROVED** [review]

**Pull request merged**